src/components/text-input.tsx

```diff
@@ -13,20 +12,72 @@ const StyledInput = styled.input`
   border: none;
   background: none;
   width: 100%;
+  outline: none;
+
+  &:disabled {
+    color: rgb(133, 133, 133);
+  }
+`;
+
+const StyledTextInput = styled.input`
+  ${commonInputStyles}
+  font-family: ${(props) => props.theme.fontFamily};
+`;
+
+const StyledTextArea = styled.textarea`
+  ${commonInputStyles}
+  font-family: ${(props) => props.theme.fontFamily};
+  flex: 1 1 auto;
+  min-height: 60px;
 `;
 
 interface TextInputProps {
   placeholder?: string;
+  isMultiline?: boolean;
+  isDisabled?: boolean;
   value?: string;
-  onChange: React.ChangeEventHandler<HTMLInputElement>;
+  onChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
+  onSubmit?: () => void;
 }
 
 export const TextInput: React.VoidFunctionComponent<TextInputProps> = ({
   placeholder,
+  isMultiline,
+  isDisabled,
   value,
   onChange,
+  onSubmit,
 }) => {
-  return (
-    <StyledInput placeholder={placeholder} onChange={onChange} value={value} />
+  const handleSubmit = (event: React.KeyboardEvent<Element>) => {
+    if (!onSubmit) {
+      return;
+    }
+    if (!event.ctrlKey) {
+      return;
+    }
+    if (event.key !== "Enter") {
+      return;
+    }
+
+    onSubmit();
+  };
```

Comment:
```suggestion
  const handleKeyDown: React.KeyboardEventHandler = (event) => {
    if (event.ctrlKey && event.key === "Enter") {
      event.preventDefault();
      onSubmit?.();
    }
  };
```

If a local function starts with `handle*`, we can use it directly in JSX: `onKeyDown={handleKeyDown}`. Ideally, we can avoid inline functions in JSX as they make it more sparse and hence harder to read.

Reply:
Done

Reply:
Thanks! Going forward, we can also use

```ts
  const handleX: React.KeyboardEventHandler = (event) => {
```

instead of 

```ts
  const handleX = (event: React.KeyboardEvent<Element>) => {
```

If there is a type available for the whole handler, we get checks for all arguments as well as the return type.